### PR TITLE
WT-7389 Remove on positioned tiered cursor should leave cursor positioned

### PR DESCRIPTION
**Pull request by keitharnoldsmith**
Rather than always using `cursor->insert()` to store the tombstone, we should use `insert()` or `update()` according to whether we need to leave the cursor positioned (i.e., whether it was positioned at the time of the `remove()`). This happens in `curtiered_put()` which is the common code for inserting something into the tables under the tiered cursor.

This change basically copies the relevant code from {{clam_put()}}.  After puzzling over what to do for a reserve operation that wants an unpositioned cursor, I realized that's impossible give the current implementation and API. So I've added an ASSERT that this doesn't happen.

### CODE REVIEW

File: src/tiered/tiered_cursor.c [hunk 1]
```diff
@@ -811,14 +812,15 @@ __curtiered_put(WT_CURSOR_TIERED *curtiered, const WT_ITEM *key, const WT_ITEM *
...
-        WT_RET(primary->insert(primary));
-    }
 
-    return (0);
+    /* Our API always leaves the cursor positioned after a reserve call. */
+    WT_ASSERT(CUR2S(curtiered), !reserve || position);
+    func = primary->insert;
+    if (position)
+        func = reserve ? primary->reserve : primary->update;
+    if (func != primary->reserve)
```
Comment (ddanderson) [inline src/tiered/tiered_cursor.c:821]: This might read a little easier as:
```
if (!reserve)
```

**Review by ddanderson — APPROVED** [review]
LGTM, one small suggestion.

**Review by sueloverso — APPROVED** [review]
LGTM - I agree with Don's suggestion.

**Comment by keitharnoldsmith** [comment]
The one Evergreen failure is a known issue: WT-7250.

**Pull request merged**